spec/features/qing_group_form_spec.rb

```diff
@@ -53,4 +53,49 @@
     within(".form-items") { expect(page).to have_content("New Group Name") }
   end
 
+  scenario "add nested groups to a form" do
+    outer_name = "Outer Group"
+    middle_name = "Middle Group"
+    inner_name = "Inner Group"
+
+    visit(edit_form_path(form, locale: "en", mode: "m", mission_name: get_mission.compact_name))
+    create_group(outer_name)
+    create_group(middle_name)
+    create_group(inner_name)
+
+    click_button("Save") # save form
+
+    # Uses DOM manipulation to simulate moving items in the draggable list.
+    # Actually using drag_to doesn't work.
+    move_items_js = <<~SCRIPT
+      function itemLi(name) {
+        return $('li .inner:contains(' + name + ')').parent();
+      }
+      function moveItem(itemName, newParentName) {
+        var newParent = itemLi(newParentName);
+        var newOl = $('<ol class=\"item-list ui-sortable\"></ol>');
+        newParent.append(newOl);
+        var item = itemLi(itemName);
+        item.detach().appendTo(newOl);
+        ELMO.formItemsView.draggable.drop_happened(null, {item: item});
+      }
+      moveItem('#{middle_name}', '#{outer_name}');
+      moveItem('#{inner_name}', '#{middle_name}');
+    SCRIPT
+    execute_script(move_items_js)
+
```

Comment:
I think we should actually have a 'Save' in here and then return to the page to ensure persistence happened.

Refreshing the page would work too I think since it saves via AJAX but might as well do 'Save' since that's what the user would do? 

Sorry if I wasn't clear about this earlier.